src/util/optional.h

```diff
@@ -38,4 +38,21 @@ typedef nonstd::bad_optional_access bad_optional_accesst;
 
 using nonstd::nullopt;
 
+/// Similar to optionalt::value but in case of empty optional, generates an
+/// invariant failure instead of throwing an exception.
+template <typename T>
+T &value(optionalt<T> &opt)
```

Comment:
That's a very generic word to introduce into the top-level namespace. How about `get_optional_value`?

Reply:
I don't know, optional is already in the signature so it is redundant and "get" is unnecessary. I don't see a problem with having it in the global namespace.

Reply:
My intent is it should be possible to read the code without navigating to the declaration to find out what that is. `value(e)` leaves ambiguous whether I'm retrieving a value from `e` or creating one that wraps `e`. If I remove that ambiguity with `get_value(e)` then it's clear in the context `optionalt<int> e = f(); auto i = get_value(e);`, but not `get_value(f())` where the `optionalt` type is not visible.

Reply:
Don't we have the same problem with `f().value()`? 
With `get_optional_value` I would expect the result to be an optional.

Reply:
OK, let's drop the `optional`, I still think `get_` is necessary to distinguish creation from retrieval though. @NathanJPhillips notes the name really ought to make clear this might throw, so `get_value_or_abort`?

Reply:
@smowton done